src/main/java/io/vertx/codegen/generators/dataobjecthelper/DataObjectHelperGen.java

```diff
@@ -99,48 +97,49 @@ private void genToJson(String visibility, boolean inheritConverter, DataObjectMo
             }
           }
         } else {
-          switch (propKind) {
-            case API:
-              if (prop.getType().getName().equals("io.vertx.core.buffer.Buffer")) {
-                genPropToJson("java.util.Base64.getEncoder().encodeToString(", ".getBytes())", prop, writer);
+          DataObjectInfo dataObject = prop.getType().getDataObject();
+          if (dataObject != null) {
+            if (dataObject.isSerializable()) {
+              String m;
+              MapperInfo mapperInfo = dataObject.getSerializer();
+              String match;
+              switch (mapperInfo.getKind()) {
+                case SELF:
+                  m = "";
+                  match = ".toJson()";
+                  break;
+                case STATIC_METHOD:
+                  m = mapperInfo.getQualifiedName() + "." + String.join(".", mapperInfo.getSelectors()) + "(";
+                  match = ")";
+                  break;
+                default:
+                  throw new UnsupportedOperationException();
               }
-              break;
-            case ENUM:
-              genPropToJson("", ".name()", prop, writer);
-              break;
-            case JSON_OBJECT:
-            case JSON_ARRAY:
-            case OBJECT:
-              genPropToJson("", "", prop, writer);
-              break;
-            case DATA_OBJECT:
-              DataObjectTypeInfo dataObjectType = ((DataObjectTypeInfo)prop.getType());
-              if (dataObjectType.isSerializable()) {
-                String m;
-                MapperInfo mapperInfo = dataObjectType.getSerializer();
-                String match;
-                switch (mapperInfo.getKind()) {
-                  case SELF:
-                    m = "";
-                    match = ".toJson()";
-                    break;
-                  case STATIC_METHOD:
-                    m = mapperInfo.getQualifiedName() + "." + String.join(".", mapperInfo.getSelectors()) + "(";
-                    match = ")";
-                    break;
-                  default:
-                    throw new UnsupportedOperationException();
+              genPropToJson(m, match, prop, writer);
+            } else {
+              return;
+            }
+          } else {
+            switch (propKind) {
+              case API:
```

Comment:
Now we can remove this right? `Buffer` will be a `DataObject`, hence it can't fall in this switch case

Reply:
yes, we'll do that in another PR. Now Buffer could be a data object but it is not yet.